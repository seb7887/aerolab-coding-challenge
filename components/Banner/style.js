import styled from 'styled-components';

export const Content = styled.div`
  min-height: 400px;
  margin: 1rem -1.5rem;
  background-size: cover;
  background-position-y: center;
  background-position-x: right;
  background-image: url('/static/images/header-x2.png');
  background-attachment: fixed;
  display: flex;
  align-items: flex-end;
  position: relative;
  perspective: 1px;

  .title {
    z-index: 2;
    font-size: 2rem;
    font-weight: 1400;
    color: ${props => props.theme.white};
    padding: 0 3rem;
  }

  .overlay {
    z-index: 1;
    width: 100%;
    height: 100%;
    position: absolute;
    background-color: ${props => props.theme.orange};
    opacity: 0.65;
  }

  @media (${props => props.theme.viewport.sm}) {
    min-height: 150px;
    margin: 1rem -0.8rem;

    .title {
      font-size: 1.5rem;
      padding: 0 1.5rem;
    }
  }
`;
