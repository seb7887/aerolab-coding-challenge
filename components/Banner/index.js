import React from 'react';

import { Content } from './style';

const Banner = ({ children }) => (
  <Content>
    <h1 className='title'>{children}</h1>
    <div className='overlay' />
  </Content>
);

export default Banner;
