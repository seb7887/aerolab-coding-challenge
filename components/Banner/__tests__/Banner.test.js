import { render } from 'react-testing-library';

import Banner from '../index';

const renderBanner = () => render(<Banner>Electronics</Banner>);

describe('Banner', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = renderBanner();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display Electronics title', () => {
    const { getByText } = renderBanner();

    expect(getByText(/electronics/i)).toBeTruthy();
  });
});
