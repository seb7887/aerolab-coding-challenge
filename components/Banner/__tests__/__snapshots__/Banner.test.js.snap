// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="style__Content-ya8ux5-0 kwQAgQ"
  >
    <h1
      class="title"
    >
      Electronics
    </h1>
    <div
      class="overlay"
    />
  </div>
</DocumentFragment>
`;
